src/ProjectService.Business/Commands/Image/CreateImageCommand.cs

```diff
@@ -12,123 +16,125 @@
+
+      return null;
+    }
+
+    public CreateImageCommand(
+      IImageRepository repository,
+      IRequestClient<ICreateImagesRequest> rcImages,
+      ILogger<CreateImageCommand> logger,
+      IAccessValidator accessValidator,
+      IHttpContextAccessor httpContextAccessor,
+      IDbImageMapper dbProjectImageMapper,
+      ICreateImageValidator validator,
+      IUserRepository userRepository)
+    {
+      _repository = repository;
+      _rcImages = rcImages;
+      _logger = logger;
+      _accessValidator = accessValidator;
+      _httpContextAccessor = httpContextAccessor;
+      _dbProjectImageMapper = dbProjectImageMapper;
+      _validator = validator;
+      _userRepository = userRepository;
+    }
 
-            if (!_accessValidator.HasRights(Rights.AddEditRemoveProjects))
-            {
-                _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
+    public OperationResultResponse<List<Guid>> Execute(CreateImageRequest request)
+    {
+      OperationResultResponse<List<Guid>> response = new();
 
-                response.Status = OperationResultStatusType.Failed;
-                response.Errors.Add("Not enough rights.");
+      Guid userId = _httpContextAccessor.HttpContext.GetUserId();
+      if (!_accessValidator.HasRights(Rights.AddEditRemoveProjects)
+        && !(request.ImageType == ImageType.Task && _userRepository.AreUserProjectExist(request.EntityId, userId)))
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
 
-                return response;
-            }
+        response.Status = OperationResultStatusType.Failed;
+        response.Errors.Add("Not enough rights.");
 
-            if (!_validator.ValidateCustom(request, out List<string> errors))
-            {
-                _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+        return response;
+      }
 
-                response.Status = OperationResultStatusType.Failed;
-                response.Errors.AddRange(errors);
+      if (!_validator.ValidateCustom(request, out List<string> errors))
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
 
-                return response;
-            }
+        response.Status = OperationResultStatusType.Failed;
+        response.Errors.AddRange(errors);
 
-            List<Guid> imagesIds = CreateImage(
-                request.Images,
-                _httpContextAccessor.HttpContext.GetUserId(),
-                response.Errors);
+        return response;
+      }
 
-            if (response.Errors.Any())
-            {
-                response.Status = OperationResultStatusType.Failed;
-                _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+      List<Guid> imagesIds = CreateImage(
+        request.Images,
+        _httpContextAccessor.HttpContext.GetUserId(),
```

Comment:
you have userId the variable in 95 string

Reply:
done